dropbox/oauth.py

```diff
@@ -108,14 +120,22 @@ def __repr__(self):
 
 class DropboxOAuth2FlowBase(object):
 
-    def __init__(self, consumer_key, consumer_secret, locale=None, token_access_type='legacy'):
+    def __init__(self, consumer_key, consumer_secret, locale=None, token_access_type='legacy',
+                 scope_list=None, include_granted_scopes=None):
+        if scope_list is not None:
+            assert isinstance(scope_list, list), \
+                "Scope list must be of type list"
```

Comment:
Also worth asserting that len(scope) > 0. 

Passing scope="" is really an edgecase where we should treat it the same as None. (We can't error back 400 since there is already some legacy apps doing that)

This behavior is confusing so I want to prevent anyone mistakenly use it.

Reply:
Ill change the assert to
`assert not scope or isinstance(scope, list)` to catch empty strings and empty lists

Reply:
Oh I mean it's totally fine to force it to be a list.

What I'm trying to say is: In your following code, `[]` will get translated to "" in uri param. We don't want that to happen. We want to explicitly assert len(scope) > 0 here.